homeassistant/components/geniushub/__init__.py

```diff
@@ -178,8 +179,16 @@ async def async_update(self, now, **kwargs) -> None:
         """Update the geniushub client's data."""
         try:
             await self.client.update()
-        except aiohttp.ClientResponseError as err:
-            _LOGGER.warning("Update failed, message is: %s", err)
+            if self._connect_error:
+                self._connect_error = False
+                _LOGGER.warning("Connection to geniushub re-established")
+        except (
+            aiohttp.ClientResponseError,
+            aiohttp.client_exceptions.ClientConnectorError,
+        ) as err:
+            if not self._connect_error:
+                self._connect_error = True
+                _LOGGER.warning("Update failed, message is: %s", err)
```

Comment:
Great little PR! 

Only minor point: the 1st warnings is about a 'Connection' & the second is about 'Update'; can the second message say: `"Connection to geniushub failed (unable to update), message is: %s"` 

Other than that, I am happy for this to be merged, thanks.

Reply:
I've updated the message in line with David's suggestion.